### PR TITLE
next release v1.1.0

### LINKED ISSUE #9: better custom completion interface
- current: `add_argument(dest, choices=shtab.DIR, metavar=dest)`
  + `metavar` required to stop `dest` being wrapped in `{}`
  + custom magic object required, where `shtab.DIR = list(shtab.Choice("directory"))`
- proposed: `add_argument(dest).complete = shtab.DIR`
  + `shtab.DIR = {"zsh": "_files", "bash": "compgen -d -S '/' --"}`
  + easier to explain & maintain; understand & extent

### PR DESCRIPTION
**Pull request by casperdcl**
- add new custom-complete API `add_argument().complete = shtab.FILE or shtab.DIR or dict(zsh=..., bash=...)` (#9)
  + support `bash`
  + support `zsh`
- update documentation
  + update `README.rst`
  + update `examples/*.py`
  + remove all mention of old `choices` API
- misc CI framework updates
  + fix autogenerated changelog missing info
  + fix tags not triggering CI

---

- fixes #9

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -223,28 +223,26 @@ Add direct support to scripts for a little more configurability:
...
-            choices=shtab.Required.DIRECTORY,  # directory tab completion
-            default=".",
+            "-s",
+            "--print-completion-shell",
+            choices=["bash", "zsh"],
+            help="prints completion script",
         )
+        # "file" & "directory" tab complete
+        parser.add_argument("file", nargs="?").complete = "file"
+        parser.add_argument("--dir", default=".").complete = "directory"
```
Comment (efiop) [inline README.rst:233]: This made me stop and think wheather or not this is a regex that should match by this name or what it was before :) shtab.FILE/DIR felt nicer, tbh.

File: shtab/__init__.py [hunk 2]
```diff
@@ -145,13 +137,25 @@ def get_bash_commands(root_parser, root_prefix, choice_functions=None):
...
         choice_type2fn.update(choice_functions)
 
     fd = io.StringIO()
     root_options = []
 
+    def get_optional_actions(parser):
+        """Flattened list of all `parser`'s optional actions."""
+        return sum(
+            (
+                # TODO: if hasattr(opt, "complete")?
```
Comment (efiop) [inline shtab/__init__.py:151]: Could create an issue instead of TODO, but your call.

Comment (shcheklein) [reply shtab/__init__.py:151]: 👍 if we keep TODO it should include a link to the proper ticket at least. And ticket should link to it.

File: README.rst (hunk 1 above)
Comment (casperdcl) [reply README.rst]: sure; added

File: shtab/__init__.py (hunk 2 above)
Comment (casperdcl) [reply shtab/__init__.py:153]: this TODO was never implemented (even in DVC's original scripts).

- add_argument("positional").complete = shtab.FILE
  + zsh: works
  + bash: didn't work in DVC, works now with shtab
- add_argument("--optional").complete = shtab.FILE
  + zsh: works
  + bash: never worked

Could open an issue...

Comment (casperdcl) [reply shtab/__init__.py:153]:
```suggestion
```

Comment (casperdcl) [reply shtab/__init__.py]: opened #11

**Pull request merged**